### PR TITLE
Remove src/backend

### PR DESCRIPTION
**Pull request by TueeNguyen**
## Issue This PR Addresses
Part of #2777 
- I'm removing the code here
- Will change the docs after this PR

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
- Remove most of backend code

## Steps to test the PR
- `cp config/env.development .env`
- `pnpm services:start parser studio db kong rest meta`
- `pnpm db:init`

- I guess if `pnpm services:start` run normally that's fine
- Need to keep in mind if other parts of telescope use anything from legacy backend
- Need to merge and test on staging to know exactly what will happen

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by TDDR** [comment]
I did a `gh pr checkout 3413` and then `pnpm services:start` and got this error 
image

Did I do something wrong?

**Comment by TueeNguyen** [comment]
@TDDR Ah they timed out, can you try again?

**Review by TDDR — APPROVED** [review]
Tested out on local and all seems to be working. I'm not sure of the greater ramifications of removing all this code but it looks to be working.

File: package.json [hunk 1]
```diff
@@ -48,46 +46,10 @@
...
-    "passport-saml": "3.2.1",
-    "pino": "7.8.0",
-    "pino-elasticsearch": "6.2.0",
-    "pino-http": "6.6.0",
-    "pino-pretty": "7.5.3",
-    "rss-parser": "3.12.0",
-    "sanitize-html": "2.6.1",
-    "set-interval-async": "2.0.3",
-    "stoppable": "1.1.0"
+    "supertest": "6.1.6"
```
Comment (rclee91) [inline package.json:52]: `supertest` is already in `devDependencies` (line 118), was the one under `dependencies` added by accident?

File: docker/development.yml [hunk 2]
```diff
@@ -20,8 +20,6 @@ services:
       args:
         # Web front-end URL
         - WEB_URL=${WEB_URL}
-        # Telescope 1.0 API URL
-        - API_URL=${API_URL}
```
Comment (TueeNguyen) [inline docker/development.yml:24]: Figure out if `API_URL` is used

**Review by JerryHue — DISMISSED** [review]
I went through all files with Tue on a meeting, so I am sure that this can be approved now. We have to be aware, though, that we modified the CI workflows, so when we merge, we gotta be attentive to the build status.

File: config/env.development [hunk 3]
```diff
@@ -187,22 +187,6 @@ DEPENDENCY_DISCOVERY_URL=http://localhost/v1/dependency-discovery
 # NODE_ENV should be one of "development" or "production"
 NODE_ENV=development
 
-# PORT is the port used by the web server
-PORT=3000
-
-# API_URL is the URL of the Telescope Web API server.  Change this to
```
Comment (humphd) [inline config/env.development:193]: The only thing we lose by doing this is the ability to work on the front-end locally, but use the staging server as your back-end.  Probably that's OK, and we should migrate to using GitPod or EC2 cloud instances instead, so you can run everything at once.

File: config/env.development [hunk 4]
```diff
@@ -187,22 +187,6 @@ DEPENDENCY_DISCOVERY_URL=http://localhost/v1/dependency-discovery
...
-#
-#   (local development)
-#   API_URL=http://localhost:3000
-#   (our staging server)
-#   API_URL=https://dev.telescope.cdot.systems
-#   (our production server)
-#   API_URL=https://telescope.cdot.systems
-API_URL=http://localhost:3000
-
 # PROXY_FRONTEND=1 will allow proxying the Next dev server (http://localhost:8000)
```
Comment (humphd) [inline config/env.development:190]: You can get rid of this next block too (`PROXY_FRONTEND=`) is only used in the backend.

File: config/env.development [hunk 5]
```diff
@@ -187,22 +187,6 @@ DEPENDENCY_DISCOVERY_URL=http://localhost/v1/dependency-discovery
 # NODE_ENV should be one of "development" or "production"
```
Comment (humphd) [inline config/env.development:187]: I wonder if we can remove this entire `Telescope 1.0 Legacy Environment` block?  Is anything in here needed?

File: docker/production.yml [hunk 6]
```diff
@@ -13,69 +13,7 @@ version: '3'
 
 services:
   ##############################################################################
-  # Telescope 1.0: Front-end and Legacy (pre-microservices) Back-end
-  ##############################################################################
-  telescope:
```
Comment (humphd) [inline docker/production.yml:18]: Bye-bye `telescope` container, you were good to us.

File: package.json [hunk 7]
```diff
@@ -13,8 +13,8 @@
     "build": "pnpm turbo run build",
     "dev": "pnpm dev --prefix src/web/app --",
     "develop": "pnpm dev",
-    "eslint": "TIMING=1 eslint --config .eslintrc.js --ignore-path .gitignore \"./src/backend/**/*.js\" \"./test/**/*.js\"",
-    "eslint-fix": "eslint --config .eslintrc.js --ignore-path .gitignore \"./src/backend/**/*.js\" \"./test/**/*.js\" --fix",
+    "eslint": "TIMING=1 eslint --ignore-path .gitignore",
```
Comment (humphd) [inline package.json:16]: If we're removing the `eslintrc` from the root, do we need these?  cc @menghif

File: src/web/app/src/config.ts [hunk 8]
```diff
@@ -24,31 +21,26 @@ const title = `Telescope`;
 const description = `A tool for tracking blogs in orbit around Seneca's open source involvement`;
 const author = `SDDS Students and Professors`;
 const keywords = `Telescope, Seneca College, Open Source, Programming, Software Development, Blogs, Education, CDOT, SDDS, Students, Professors, Learning, Teaching`;
-const image = `${telescopeUrl}/logoImages/apple-splash-1334-750.jpg`;
+const image = `/logo.svg`;
```
Comment (humphd) [inline src/web/app/src/config.ts:24]: Is an SVG compatible with a JPG in the ways we use this?

**Review by humphd — CHANGES_REQUESTED** [review]
Nice work.  A few questions.

File: package.json (hunk 7 above)
Comment (menghif) [reply package.json:16]: We can remove both `eslint` and `eslint-fix`.

File: package.json [hunk 9]
```diff
@@ -13,8 +13,8 @@
...
+    "eslint-fix": "eslint --ignore-path .gitignore --fix",
     "lint": "pnpm turbo run lint && pnpm eslint",
```
Comment (menghif) [inline package.json:18]: Change this to: `"lint": "pnpm turbo run lint",`

**Review by menghif — CHANGES_REQUESTED** [review]

File: src/web/app/src/config.ts (hunk 8 above)
Comment (TueeNguyen) [reply src/web/app/src/config.ts:24]: So the `image` is actually being used here
https://github.com/Seneca-CDOT/telescope/blob/6046273fba4b89ce6281f7a34cafbdec7ef222dc/src/web/app/src/pages/_document.tsx#L6-L36
We tested it out by starting the services and there wasn't a problem

File: config/env.development (hunk 5 above)
Comment (TueeNguyen) [reply config/env.development:187]: Parser still uses these ones 
```yml
      - NODE_ENV
      - PARSER_PORT
      - ELASTIC_MAX_RESULTS_PER_PAGE
      - ELASTIC_URL=http://elasticsearch
      - ELASTIC_PORT=9200
```

Comment (TueeNguyen) [reply config/env.development:187]: Seems like only `parser` uses those variables. I can do some detective work, move in-use variables to parser section

**Comment by humphd** [comment]
This needs a rebase.  Also, where are we at with the changes it makes?  Can we do this soon?

**Comment by TueeNguyen** [comment]
I've removed some config code in `/config`. It seems like all of `Telescope 1.0 Legacy Environment` variables are used. I changed it to `Environment` and rebased

**Review by humphd — CHANGES_REQUESTED** [review]
This looks right, but running it locally, the parser crashes with a missing service role key for Supabase:

```
{"level":50,"time":1649962783990,"pid":7,"hostname":"5b99780d3f7e","err":{"type":"Error","message":"Invalid authentication credentials","stack":"Error: Invalid authentication credentials\n    at getAllFeeds (/app/src/utils/supabase.js:21:13)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async processAllFeeds (/app/src/parser.js:62:19)"},"msg":"Error queuing feeds"}
```

Make sure you pass this into the environment for the parser container in docker-compose.yml

**Comment by TueeNguyen** [comment]
@humphd Ah, I think it's failing because parser is using staging supabase and the `SERVICE_ROLE_KEY` is incorrect

**Comment by humphd** [comment]
So then we can't use staging like this, since we'll never have the service role key, and depend on it.  Can you switch to use the local one?

**Comment by TueeNguyen** [comment]
I wonder how this [PR](https://github.com/Seneca-CDOT/telescope/pull/3363) ran normally

**Comment by TueeNguyen** [comment]
There's a problem if we use local supabase, we won't have all the feeds in the table already.

**Comment by humphd** [comment]
Can we solve that problem in a follow-up?

**Comment by TueeNguyen** [comment]
Sure!

**Comment by TueeNguyen** [comment]
Changed `SUPABASE_URL` to http://kong:8000/` and rebased.

**Comment by TDDR** [comment]
I think if you fix the merge conflicts this will be good to merge.

**Comment by rclee91** [comment]
Just want to note that running on the current master with `pnpm services:start` (with no backend), has no posts for me. 
I'll need time to test this branch specifically. 

Unless it might be a me problem. If someone else can test.

**Comment by rclee91** [comment]
Tested this PR and for me, I'm still not getting any posts with `pnpm:services start`
On `parser` it connects to `ElasticSearch` and that's it.

**Comment by TueeNguyen** [comment]
@rclee91 did you run db migrate? The supabase database might be empty

**Comment by rclee91** [comment]
> @rclee91 did you run db migrate? The supabase database might be empty. Acutally, lemme update the test steps

Before I do that. In the case my potato runs out of storage, what do I run to get rid of the database

**Comment by TueeNguyen** [comment]
@rclee91 haha don't worry, you can remove the volume in docker desktop.

**Comment by TDDR** [comment]
When I followed your testing instructions, I got this on the second step.
image

**Comment by TueeNguyen** [comment]
@TDDR I think you can skip that step if it's not successful

**Comment by DukeManh** [comment]
I updated the instruction a bit, removed `prisma pull` and added `cp env.example .env.

**Comment by TDDR** [comment]
I also updated the testing by adding another `cp env.example .env` after changing to the `tools/migrate` folder. After that I was successfully migrate and sync. Everything appears to be working.

**Review by TDDR — DISMISSED** [review]

**Comment by TueeNguyen** [comment]
Thanks guys

**Comment by TueeNguyen** [comment]
@humphd It's on the top under `Steps to test this PR` section

**Review by humphd — APPROVED** [review]
> I guess if pnpm services:start run normally that's fine

This won't work in the steps above: the parser will wait for the queue to empty before trying to re-load the feeds, which will never happen since the queue is already empty.  You have to restart the `parser` container and then it runs.  That's a bug to fix in a follow-up.

I think this is working as expected.

I want to take the steps above and automate them with a script.  It shouldn't be this hard to get the local environment to "boot."  We need an issue filed on that too.

Please file these issues and this is good to go.

**Review by HyperTHD — DISMISSED** [review]

**Review by humphd — APPROVED** [review]
If this passes CI, r+ from me.

**Review by JiaHua-Zou — APPROVED** [review]

**Review by JiaHua-Zou — DISMISSED** [review]

**Review by Kevan-Y — DISMISSED** [review]
Tested locally, LGTM

**Comment by humphd** [comment]
Conflicts need to be resolved.  Let's get this finished and landed so we can test, @TueeNguyen

**Comment by TueeNguyen** [comment]
Conflicts resolved!

**Review by humphd — APPROVED** [review]
If this passes CI, land it.

**Review by TDDR — DISMISSED** [review]

File: docker/docker-compose.yml [hunk 10]
```diff
@@ -265,9 +265,7 @@ services:
       # next.js needs build-time access to a number of API URL values, forward as ARGs
       args:
         # Web front-end URL
-        - WEB_URL
-        # Telescope 1.0 API URL
-        - API_URL
+        - WEB_URL=${WEB_URL}
```
Comment (Kevan-Y) [inline docker/docker-compose.yml:268]: I think `WEB_UR` is enough, we can just forward the args.

```suggestion
        - WEB_URL
```

**Review by Kevan-Y — CHANGES_REQUESTED** [review]

**Review by joelazwar — APPROVED** [review]

**Review by Kevan-Y — DISMISSED** [review]

**Comment by TDDR** [comment]
Do we know why this is failing CI?

**Review by TDDR — DISMISSED** [review]

**Comment by TueeNguyen** [comment]
@TDDR it's good now

**Review by JiaHua-Zou — DISMISSED** [review]

**Review by joelazwar — DISMISSED** [review]

**Review by DukeManh — APPROVED** [review]

**Review by JiaHua-Zou — DISMISSED** [review]

**Review by JiaHua-Zou — APPROVED** [review]

**Review by sirinoks — APPROVED** [review]

**Review by joelazwar — APPROVED** [review]

**Pull request merged**